OMCompiler/Compiler/NFFrontEnd/NFConnectEquations.mo

```diff
@@ -829,12 +829,12 @@ algorithm
     rel_exp := Expression.IF(
       Expression.RELATION(flow_exp, op, Expression.REAL(0.0)),
       instream_exp, stream_exp);
-  end if;
 
-  // actualStream(stream_var) = smooth(0, if flow_var > 0 then inStream(stream_var)
-  //                                                      else stream_var);
-  exp := Expression.CALL(Call.makeTypedCall(NFBuiltinFuncs.SMOOTH,
-    {DAE.INTEGER(0), rel_exp}, Expression.variability(rel_exp)));
+    // actualStream(stream_var) = smooth(0, if flow_var > 0 then inStream(stream_var)
+    //                                                      else stream_var);
+    exp := Expression.CALL(Call.makeTypedCall(NFBuiltinFuncs.SMOOTH,
```

Comment:
don't you have to set exp := rel_exp; in the other cases? or directly define exp instead of rel_exp?

Reply:
Yep, fixed! Maybe I should actually try to compile the code before pushing it...